ChromePlugin/background.js

```diff
@@ -40,48 +58,53 @@ function FormatDuration(d) {
   return Math.floor(d / divisor[0]) + ":" + pad(Math.floor((d % divisor[0]) / divisor[1]));
 }
 
-function Update(dateTime, tabId, url) {
+function Update(dateTimeStart, dateTimeEnd, tabId, url, reffererUrl) {
   if (!url) { return; }
   //alert(url);
   if (tabId in History) {
     if (url == History[tabId][0][1]) { return; }
   } else { History[tabId] = []; }
 
   // Add to the beginning of the array.
-  History[tabId].unshift([dateTime, url]);
+  History[tabId].unshift([dateTimeStart, url]);
 
   // Clean History so it doesnt explode =)
   var history_limit = parseInt(localStorage["history_size"]);
-  if (! history_limit) { history_limit = 23; }
+  if (!history_limit) { history_limit = 23; }
   while (History[tabId].length > history_limit) { History[tabId].pop(); }
 
-  chrome.browserAction.setBadgeText({ 'tabId': tabId, 'text': '0:00'});
-  chrome.browserAction.setPopup({ 'tabId': tabId, 'popup': "popup.html#tabId=" + tabId});
+  chrome.browserAction.setBadgeText({ 'tabId': tabId, 'text': '0:00' });
+  chrome.browserAction.setPopup({ 'tabId': tabId, 'popup': "popup.html#tabId=" + tabId });
 
-  SendDataToServer();
+  /* Sends the new site data and the previous site, including end datetime */
+  let newSite = getUsageData(tabId, 0);
+  let PreviousSite = getUsageData(tabId, 1);
+  let usageData = PreviousSite != null ? [newSite, PreviousSite] : [newSite];
+  SendDataToServer(usageData);
```

Comment:
review your code here. Naming conventions caps lower. and semantic naming. Site vs Url